routers/api/v1/api.go

```diff
@@ -266,287 +266,295 @@ func mustAllowPulls(ctx *context.Context) {
 	}
 }
 
-// RegisterRoutes registers all v1 APIs routes to web application.
-// FIXME: custom form error response
-func RegisterRoutes(m *macaron.Macaron) {
+func apiRoutes(m *macaron.Macaron) {
```

Comment:
I would add a `ghCompat bool` argument here, and `if !ghCompat { ... }` around any custom endpoints we have